src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -54,87 +63,75 @@ const defaultProps = {
     bankAccountList: [],
     cardList: [],
     isLoadingPayments: false,
+    enableSelection: false,
+    selectedAccountID: '',
+    filterType: '',
     isAddPaymentMenuActive: false,
 };
 
 class PaymentMethodList extends Component {
     constructor(props) {
         super(props);
-
+        this.state = {
+            selectedAccountID: this.props.selectedAccountID,
+        };
         this.renderItem = this.renderItem.bind(this);
+        this.selectPaymentMethod = this.selectPaymentMethod.bind(this);
+        this.createPaymentMethodList = this.createPaymentMethodList.bind(this);
     }
 
     /**
      * Take all of the different payment methods and create a list that can be easily digested by renderItem
-     *
      * @returns {Array}
      */
     createPaymentMethodList() {
-        const combinedPaymentMethods = [];
+        let paymentMethods = [];
+        if (_.isEmpty(this.props.filterType)) {
+            paymentMethods = PaymentUtils.getPaymentMethodsList(this.props.bankAccountList, this.props.cardList, this.props.payPalMeUsername);
+        } else {
+            paymentMethods = PaymentUtils.getPaymentMethodsList(
+                this.props.filterType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? this.props.bankAccountList : [],
+                this.props.filterType === CONST.PAYMENT_METHODS.DEBIT_CARD ? this.props.cardList : [],
+                !this.props.filterType ? this.props.payPalMeUsername : '',
+            );
+        }
```

Comment:
I think it would be much clearer if we just filtered out the things that we do/don't want after getting the payment methods. All of these ternaries make this logic difficult to follow.